skimage/filters/thresholding.py

```diff
@@ -294,7 +299,7 @@ def _validate_image_histogram(image, hist, nbins=None):
             counts, bin_centers = counts[start:end], bin_centers[start:end]
     else:
         counts, bin_centers = histogram(
-                image.ravel(), nbins, source_range='image'
+                image.ravel(), nbins, source_range='image', normalize=normalize
```

Comment:
```suggestion
                image.reshape(-1), nbins, source_range='image', normalize=normalize
```
:wink:

Reply:
Hi @rfezzani, I didn't touch things like that one because I wanted to be as least intrussive as possible. Also, ravel is being used in several other places, so changing this would create inconsistencies, and I felt that changing the style of the whole module was out of the scope of this PR. I see that there are [semantic differences between ravel and reshape](https://stackoverflow.com/a/28930580), do you propose to just change this entry or would you suggest changing them all? In the latter case I'd suggest to do so in a separate PR

Reply:
@andres-fr, please see the **Notes** section in `np.ravel` [doc](https://numpy.org/doc/stable/reference/generated/numpy.ravel.html) :wink:

Reply:
Thank you, yes that is aligned with the SO post. I agree with you that `histogram` wouldn't need a copy so we want to provide a flattened view of the array if we can. But I also suppose this is the case for many other uses of `ravel` in the repo. Do you think it would be an idea to open an enhancement issue on this?

Reply:
OK I dared to `grep -r "ravel" .` in the repo and quite a bit came out. Would `*.c, *.pyx...` files also be in the scope or just `*.py`? Do you know if `ravel/unravel_index` would also be affected by this? (docs don't mention anything)

Reply:
`*.c`  files are generally generated by Cython, so it doesn't seem in the scope, but `*.pyx` maybe concerned. `ravel/unravel_index` are not concerned by extra/unwanted data copy.
Thank you again for your reactivity, I will open another issue to continue this discussion there :wink:.

Reply:
please see #5555